File: ios/Runner/Model/Contract/AlbumModelContract.swift
//
//  AlbumModelContract.swift
//  Runner
//
//  Created by t k on 2024/08/12.
//

import MediaPlayer

protocol AlbumModelContract {
    var media: MPMediaItemCollection { get }

    var id: AlbumId { get }

    var name: String { get }

    var artistId: ArtistId { get }

    var artistName: String { get }

    var imageId: String { get }

    var artwork: MPMediaItemArtwork? { get }
}

File: android/app/src/main/kotlin/dev/tcode/thinmpf/model/contract/AlbumModelContract.kt
package dev.tcode.thinmpf.model.contract

import android.net.Uri
import dev.tcode.thinmpf.model.valueObject.AlbumId
import dev.tcode.thinmpf.model.valueObject.ArtistId

interface AlbumModelContract {
    val id: AlbumId
    val name: String
    val artistId: ArtistId
    val artistName: String
    val imageId: String
    val imageUri: Uri
}